nuxeo-full-ps-template.yaml

```diff
@@ -0,0 +1,1989 @@
+                                      passwordVariable: "NUXEO_CONNECT_PASSWORD"
+                                  )
+                              ]) {
+                                  unstash name: "mavenSettings"
+                                  def mavenSettings = findFiles(glob: "**/settings.xml")[0]
+                                  sh "mvn --settings ${mavenSettings.path} clean package"
+                                  if (fileExists(".nuxeo-s2i")) {
+                                      def props = readProperties file: ".nuxeo-s2i"
+                                      packagePathRegExp = props["NUXEO_PACKAGE_DIR"]
+                                  } else {
+                                      packagePathRegExp = "*(marketplace|package)-*.zip"
+                                  }
+                                  def files = findFiles(glob: "**/${packagePathRegExp}")
+                                  if (files.length == 0) {
+                                      echo "Marketplace zip file not found."
+                                  }
+                                  stash name:"marketplace", includes:"${files[0]}"
+                              }
+                          }
+                      }
+                    }
+                  node {
+                      stage("Build Image") {
+                          unstash name: "marketplace"
+                          def files = findFiles(glob: "**/${packagePathRegExp}")
+                          if (files.length == 0) {
+                              echo "Marketplace zip file not found."
+                          }
+                          if (fileExists("source")) {
+                              sh "rm -rf source"
+                          }
+                          sh "mkdir -p source"
+                          fileOperations([
+                              fileCopyOperation(
+                                  includes: "${files[0].path}",
+                                  targetLocation: "source",
+                                  flattenFiles: true,
+                              ),
+                              fileRenameOperation(
+                                  source: "source/${files[0].name}",
+                                  destination: "source/marketplace.zip"
+                              ),
+                              fileDownloadOperation(
+                                  url: "https://raw.githubusercontent.com/nuxeo-sandbox/nuxeo-s2i/${NUXEO_VERSION}/contrib/install.sh",
+                                  userName: "",
+                                  password: "",
+                                  targetLocation: "source",
+                                  targetFileName: "install.sh"
+                              )
+                          ])
+                          sh "oc start-build ${APPLICATION_NAME}-app-build --from-dir='source' -n ${project}"
```

Comment:
Is there not a way to let Jenkins build the Docker image rather that delegating to an external build? 

It would surely be better in the future when we will want to refer to the built image.

Reply:
Why not, and I tried, but I think it would take me quite of time.
So I'd prefer to keep it this way for now, and maybe improve it in a second version.